result/src/main/kotlin/com/alliander/result/Result.kt

```diff
@@ -113,6 +113,22 @@ sealed class Result<Error, Value> {
         }
     }
 
+    /**
+     * Chain two computations together for error recovery.
+     *
+     * If a computation was a [Failure], `recover` can chain the error data into an error recovery computation.
+     * Otherwise keep the [Success].
+     *
+     * @param transform Provides the transformed error
+     * @return The transformed error of a [Failure]. Keeps the data otherwise.
+     */
+    fun <T> recover(transform: (Error) -> Result<T, Value>): Result<T, Value> {
```

Comment:
This is a neat function, that fits the API perfectly.

The only concern I have is about the name. Basically it is an `andThen`, but for failures.

There are similar method pairs, e.g. `map` and `mapError`, `use` and `useError`. So I am in favor of keeping the implementation, but changing the name to `andThenError`.